simtools/layout/layout_array.py

```diff
@@ -362,40 +445,62 @@ def _load_telescope_list(self, table):
             tel = self._load_telescope_names(row)
 
             try:
-                tel.set_coordinates(
-                    "corsika",
-                    row["pos_x"] * table["pos_x"].unit,
-                    row["pos_y"] * table["pos_y"].unit,
-                )
+                if not isinstance(row["pos_x"], u.Quantity) and not isinstance(
+                    row["pos_y"], u.Quantity
+                ):
+                    tel.set_coordinates(
+                        "corsika",
+                        row["pos_x"] * table["pos_x"].unit,
+                        row["pos_y"] * table["pos_y"].unit,
+                    )
+                else:
+                    tel.set_coordinates("corsika", row["pos_x"], row["pos_y"])
             except KeyError:
                 pass
             try:
-                tel.set_coordinates(
-                    "utm",
-                    row["utm_east"] * table["utm_east"].unit,
-                    row["utm_north"] * table["utm_north"].unit,
-                )
+                if not isinstance(row["utm_east"], u.Quantity) and not isinstance(
```

Comment:
Here and all the following if statements: this is a lot of repetition and maybe we can simplify the code by:
```
tel.set_coordinates("utm", 
      assign_unit_to_quantity( row["utm_east"], table["utm_east"].unit),
      assign_unit_to_quantity( row["utm_north"], table["utm_north"].unit ))
```
and a small helper function `assign_unit_to_quantity` which does the job of all those if statements.


```

Reply:
That is a good idea. I now created not only the `_assign_unit_to_quantity` but also the `_try_set_coordinate` to avoid the excessive number of `try and excepts` and the `_try_set_altitude` for consistency. The three new functions have also unit tests now. Please have a look and let me know what you think. 

There are other places with a lot of if and try and except statements, such as  many of the _initialize fuinctions but I wouldnt further extend this PR but rather open an issue for that?

Reply:
Great, thanks (and I believe it was me introducing the repetitive code)